src/Surfnet/StepupRa/RaBundle/Security/Authentication/Provider/SamlProvider.php

```diff
@@ -92,43 +90,41 @@ public function authenticate(TokenInterface $token)
             );
         }
 
-        $raCredentials = $this->identityService->getRaCredentials($identity);
+        $profile = $this->profileService->findByIdentityId($identity->id);
 
         // if no credentials can be found, we're done.
-        if (!$raCredentials) {
+        if (!$profile->isSraa && empty($profile->authorizations) && empty($profile->management)) {
             throw new UserNotRaException(
                 'The Identity is not registered as (S)RA(A) and therefor does not have access to this application'
             );
         }
 
         // determine the role based on the credentials given
         $roles = [];
-        if ($raCredentials->isSraa) {
+        if ($profile->isSraa) {
             $roles[] = 'ROLE_SRAA';
         }
 
-        if ($raCredentials->isRaa) {
-            $roles[] = 'ROLE_RAA';
-        } else {
-            $roles[] = 'ROLE_RA';
+        // Get authorizations (use_ra/use_raa)
+        foreach ($profile->authorizations as $institution => $role) {
+            if ($role[0] == 'raa' && !in_array('ROLE_RAA', $roles)) {
+                $roles[] = 'ROLE_RAA';
+            }
+            if ($role[0] == 'ra' && !in_array('ROLE_RA', $roles)) {
+                $roles[] = 'ROLE_RA';
+            }
         }
 
-        $institutionConfigurationOptions = $this->institutionConfigurationOptionsService
-            ->getInstitutionConfigurationOptionsFor($identity->institution);
-
-        if ($institutionConfigurationOptions === null) {
-            throw new InconsistentStateException(
-                sprintf(
-                    'InstitutionConfigurationOptions for institution "%s" '
-                    . 'must exist but cannot be found after authenticating Identity "%s"',
-                    $identity->institution,
-                    $identity->id
-                )
-            );
+        // Get authorizations (select_raa)
+        if (!empty($profile->management)) {
+            foreach ($profile->management as $institution) {
+                $roles[] = 'ROLE_SELECT_RAA';
```

Comment:
For the sake of transparency; mustn't we add this new role to the role_hierarchy in security.yml? Or mention it in docs somewhere?